File: Ruby/day04/main.rb
require 'digest/md5'

def findResult(secret_key, str_to_find)
  length = str_to_find.length
  1.upto(Float::MAX) do |n|
    if Digest::MD5.hexdigest("#{secret_key}#{n}")[0...length] == str_to_find
      puts "result is #{n}"
      break
    end
  end
end

secret_key = 'ckczppom'
findResult(secret_key, '00000')
findResult(secret_key, '000000')

File: Go/day04/main.go
package main

import (
	"fmt"
	"crypto/md5"
	"strings"
	"strconv"
	"encoding/hex"
)

func main() {
	secretKey := "ckczppom"
	findResult(secretKey, "00000")
	findResult(secretKey, "000000")
}

func findResult(secretKey string, strToFind string) {
	i := 1
	for {
		hash := md5.Sum([]byte(secretKey + strconv.Itoa(i)))
		if strings.HasPrefix(hex.EncodeToString(hash[:]), strToFind) {
			fmt.Println("result is", i)
			break
		}
		i++
	}
}